webpack/assets/javascripts/react_app/components/SettingUpdateModal/components/SettingValueField.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Col, HelpBlock, FormGroup, FormControl } from 'patternfly-react';
+
+import { translate as __ } from '../../../common/I18n';
+import { arraySelection } from '../../SettingsTable/SettingsTableHelpers';
+import { renderOptions } from '../../common/forms/SelectHelpers';
+
+const SettingValueField = ({ setting, form, field }) => {
```

Comment:
Still not ideal, but I believe good enough (all the convoluted statements are hidden in this component, so would be easier to refactor in the future)